nixos/modules/services/security/step-ca.nix

```diff
@@ -0,0 +1,125 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.step-ca;
+  settingsFormat = (pkgs.formats.json { });
+in
+{
+  options = {
+    services.step-ca = {
+      enable = lib.mkEnableOption "the smallstep certificate authority server";
+      openFirewall = lib.mkEnableOption "opening the certificate authority server port";
+      settings = lib.mkOption {
+        type = lib.types.submodule {
+          freeformType = with lib.types; attrsOf anything;
+
+          options.address = lib.mkOption {
```

Comment:
`address` would also be a step-ca option. But I think the easiest way is probably to just move both out of `settings` and unconditionally override `settings.address` with `address`:`port`. Was that what you meant? (This would mean both these options are required)